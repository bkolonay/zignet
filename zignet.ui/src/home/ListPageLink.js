import React, { Component } from 'react';

class ListPageLink extends Component {

  render() {

	  const link = this.props.grouped ? 
	  	<a href={`/${this.props.suiteId}?group=true`}>Total: {this.props.totalTests}</a> :
	  	<a href={`/${this.props.suiteId}`}>Total: {this.props.totalTests}</a>;

    return (
      link
    );
  }
}

export default ListPageLink;
