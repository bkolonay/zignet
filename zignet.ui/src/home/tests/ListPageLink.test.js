import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ListPageLink from '../ListPageLink';

it('renders when not grouped', () => {
  const component = <ListPageLink grouped={false} suiteId={1} totalTests={2} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('renders when not grouped', () => {
  const component = <ListPageLink grouped={true} suiteId={1} totalTests={2} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('snapshot when not grouped', () => {
  const component = <ListPageLink grouped={false} suiteId={1} totalTests={2} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when grouped', () => {
  const component = <ListPageLink grouped={true} suiteId={1} totalTests={2} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when total tests zero', () => {
  const component = <ListPageLink grouped={false} suiteId={1} totalTests={0} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});