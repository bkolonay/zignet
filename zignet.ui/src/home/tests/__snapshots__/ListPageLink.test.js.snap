// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot when grouped 1`] = `
<a
  href="/1?group=true"
>
  Total: 
  2
</a>
`;

exports[`snapshot when not grouped 1`] = `
<a
  href="/1"
>
  Total: 
  2
</a>
`;

exports[`snapshot when total tests zero 1`] = `
<a
  href="/1"
>
  Total: 
  0
</a>
`;
